File: server/plugin/app/router/consultation.go
package router

import (
    "github.com/flipped-aurora/gin-vue-admin/server/middleware"
    "github.com/gin-gonic/gin"
)

var Consultation = new(CN)

type CN struct{}

func (r *CN) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
    {
        group := private.Group("CN").Use(middleware.OperationRecord())
        group.POST("createConsultation", apiConsultation.CreateConsultation)
        group.DELETE("deleteConsultation", apiConsultation.DeleteConsultation)
        group.DELETE("deleteConsultationByIds", apiConsultation.DeleteConsultationByIds)
        group.PUT("updateConsultation", apiConsultation.UpdateConsultation)
    }
    {
        group := private.Group("CN")
        group.GET("findConsultation", apiConsultation.FindConsultation)
        group.GET("getConsultationList", apiConsultation.GetConsultationList)
    }
    {
        group := public.Group("CN")
        group.GET("getConsultationPublic", apiConsultation.GetConsultationPublic)
        group.GET("findConsultationPublic", apiConsultation.FindConsultationPublic)
    }
}

File: web/src/plugin/app/api/consultation.js
import service from '@/utils/request'

export const createConsultation = (data) => {
  return service({ url: '/CN/createConsultation', method: 'post', data })
}

export const deleteConsultation = (params) => {
  return service({ url: '/CN/deleteConsultation', method: 'delete', params })
}

export const deleteConsultationByIds = (params) => {
  return service({ url: '/CN/deleteConsultationByIds', method: 'delete', params })
}

export const updateConsultation = (data) => {
  return service({ url: '/CN/updateConsultation', method: 'put', data })
}

export const findConsultation = (params) => {
  return service({ url: '/CN/findConsultation', method: 'get', params })
}

export const getConsultationList = (params) => {
  return service({ url: '/CN/getConsultationList', method: 'get', params })
}

export const getConsultationPublic = (params) => {
  return service({ url: '/CN/getConsultationPublic', method: 'get', params })
}

export const findConsultationPublic = (params) => {
  return service({ url: '/CN/findConsultationPublic', method: 'get', params })
}